vllm/model_executor/models/jamba.py

```diff
@@ -82,9 +86,9 @@ def __init__(self, config: JambaConfig, layer_idx):
         # doesn't allow to override it
         self.conv1d.weight.data = self.conv1d.weight.data.unsqueeze(1)
 
-        self.in_proj = MergedColumnParallelLinear(self.hidden_size,
-                                                  [self.intermediate_size] * 2,
-                                                  bias=self.use_bias)
+        self.in_proj = ColumnParallelLinear(self.hidden_size,
+                                            self.intermediate_size * 2,
+                                            bias=self.use_bias)
```

Comment:
Are you sure this doesn't need to be `MergedColumnParallelLinear`? It looks like we chunk the output in two pieces immediately after applying `self.in_proj` in the forward method. This would work with the TP==1 case but IIUC would break TP > 1 as the weight loading would be incorrect. 

We should also add a tensor parallel integration test for Jamba

Reply:
You're right, reverted the change , Added a test that verifies TP>2 works as intended.